sdlib/d/gc/bin.d

```diff
@@ -196,5 +196,46 @@ struct BinInfo {
 	}
 }
 
+// Determine whether given size class may hold appendability flag.
+bool mayAppend(ubyte sizeClass) {
+	return !isSmall(sizeClass) || (binInfos[sizeClass].slots <= 256);
+}
+
+// Determine whether given size class may hold finalization flag.
+bool mayFinalize(ubyte sizeClass) {
+	return !isSmall(sizeClass) || (binInfos[sizeClass].slots <= 128);
+}
+
+// Take given alloc size and augment if required to store the given meta flags.
+size_t sizeUpWithMeta(size_t size, bool append, bool finalize) {
+	auto sc = getSizeClass(size);
+	while ((append && !mayAppend(sc)) || (finalize && !mayFinalize(sc))) {
+		sc += 1;
+		assert(sc <= ubyte.max);
+	}
+
+	return getSizeFromClass(sc);
+}
```

Comment:
Without any use for these methods, it is once again impossible to determine if they provide the right functionality.

Nevertheless, my gut feeling is that they do not because one will have to work from the size, add the required metadata, and then work out which size class it corresponds to, which is not what this providing.

More generally, instead of trying to do everything at once, let's make something work. The refactoring with `isSmall` is one such thing. Alternatively, making just apendable, just for large classes would be a useful first step.

Reply:
@deadalnix it does in fact work, and all of the worked-out context is [here](https://github.com/snazzy-d/sdc/compare/master...dsm9000:sdc:store_alloc_size#diff-baf796be0101f02a1d56248a34f4597b719bbbeff463439f8ee962abf710eef0R150). This MR is my attempt to separate out the currently existing and tested components.